### ISSUE TITLE
swarm-frontend for bare metals

### ISSUE THREAD

**Issue by navinSing** [docker/swarm-frontends#6]
what would be the steps for bare-metals ..
want to setup local env

**Comment by amitshukla** [comment]
@navinSing can you clarify what you are looking for? E.g.: are you trying to use these compose files to setup a mesos or kubernetes cluster on your local machine without using Virtual machines?

**Comment by navinSing** [comment]
Looking for just 2 things ..
1. Setup script for the slave machine.. I mean what will be the config/
   order .. Mesos slave ... Or docker slave along with the deamon params to
   connect to consul.
2. And local docker registry for the LAN.

I spent like 3 days trying to do this without success.

It will be great if you can help me here.

Thanks

**Comment by amitshukla** [comment]
@navinSing if you look at init_swarm.sh, you should be able to use this command as a template to get swarm on the machine:
echo "----- create a machine for swarm node 5 -----"
docker-machine create -d virtualbox --swarm \
    --swarm-discovery="consul://$(docker-machine ip mh-keystore):8500" \
    --engine-opt="cluster-store=consul://$(docker-machine ip mh-keystore):8500" \
    --engine-opt="cluster-advertise=eth1:2376" \
    swarm-node-05

After this, you should be able to use the compose file to configure Mesos master, slaves etc.

Look at the docker-machine documentation for your specific scenarion. For example, this page lists how to use docker-machine to configure a machine in AWS:
https://docs.docker.com/machine/drivers/aws/

**Comment by navinSing** [comment]
@amitshukla , sorry.. let me clarify ..  i am looking for bare metals .. at this moment not aws or vagrant.

Assume that i have a linux machine. what all things i should do to join it to the swarm. I was specifically looking at mesos example.

so i will have to install swarm-agent and mesos slave .. what will be the order. What ssh keys sharing i need to do ..

Also if i can get a local docker registry it will be great..

**Comment by amitshukla** [comment]
@navinSing please read this article: https://blog.codefresh.io/docker-machine-basics/
And use the generic driver (Ubuntu server over ssh).

**Comment by navinSing** [comment]
nah, that is not what i was looking for it simply installs docker-machine on that host, looking for a swarm cluster to connect to the rest of the machines. The example doesn't use ssh keys etc..

**Comment by navinSing** [comment]
any updates on this .. please
